API/MoodisAPI/apiMoodis/apiMoodis/Controllers/EmotionController.cs

```diff
@@ -0,0 +1,95 @@
+﻿using apiMoodis.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+using System.Data.Entity;
+
+namespace apiMoodis.Controllers
+{
+    public class EmotionController : ApiController
+    {
+        public HttpResponseMessage Get()
+        {
+            using (DatabaseContext dbContext = new DatabaseContext())
+            {
+                var emotions = dbContext.Emotions.ToList();
+                if (emotions.Count() == 0)
+                {
+                    return Request.CreateErrorResponse(HttpStatusCode.NotFound, "There are no emotions");
+                }
+                else
+                {
+                    return Request.CreateResponse(HttpStatusCode.OK, emotions);
+                }
+            }
+        }
+
+        public HttpResponseMessage Get(string id)
+        {
+            using (DatabaseContext dbContext = new DatabaseContext())
+            {
+                var entity = dbContext.Emotions.FirstOrDefault(emotion => emotion.Id == id);
```

Comment:
This looks weird to me. Why are you using `FirstOrDefault` and then checking for `null`? This would be useful, if you did some additional business logic in that case. 

Since you are throwing an exception, I would suggest using:
`dbContext.Emotions.First(...)`
Or even in this case:
`dbContext.Emotions.Single(...)`

Those methods are useful, since the exception will be thrown automatically if there are no elements.

Reply:
All of the uncaught exceptions will automatically be wrapped in HTTP code 500